pkg/alertmanager/statefulset.go

```diff
@@ -223,71 +224,73 @@ func makeStatefulSetSpec(logger *slog.Logger, a *monitoringv1.Alertmanager, conf
 		return nil, fmt.Errorf("failed to parse alertmanager version: %w", err)
 	}
 
-	amArgs := []string{
-		fmt.Sprintf("--config.file=%s", path.Join(alertmanagerConfigOutDir, alertmanagerConfigEnvsubstFilename)),
-		fmt.Sprintf("--storage.path=%s", alertmanagerStorageDir),
-		fmt.Sprintf("--data.retention=%s", a.Spec.Retention),
+	amArgs := []monitoringv1.Argument{
+		{Name: "config.file", Value: path.Join(alertmanagerConfigDir, alertmanagerConfigFileCompressed)},
+		{Name: "storage.path", Value: alertmanagerStorageDir},
+		{Name: "data.retention", Value: string(a.Spec.Retention)},
 	}
 
-	if *a.Spec.Replicas == 1 && !a.Spec.ForceEnableClusterMode {
-		amArgs = append(amArgs, "--cluster.listen-address=")
-	} else {
-		amArgs = append(amArgs, "--cluster.listen-address=[$(POD_IP)]:9094")
+	clusterListenAddressArg := monitoringv1.Argument{Name: "cluster.listen-address", Value: `""`}
```

Comment:
@simonpasquier I think this works. I guess if the e2e tests pass we are i the clear.